File: examples/wealthreader.js
/**
 * Wealthreader API - Ejemplo en Node.js
 * Obtiene activos financieros de una entidad bancaria.
 * Compatible con Node.js 18+ (fetch nativo) o navegador.
 */

const API_URL = "https://api.wealthreader.com/entities/";

const payload = new URLSearchParams({
  api_key: "TU_API_KEY",
  code: "caixabank",
  token: "TU_TOKEN",
  product_types: "accounts,portfolios",
  date_from: "2024-01-01",
});

async function getEntities() {
  const response = await fetch(API_URL, {
    method: "POST",
    headers: { "Content-Type": "application/x-www-form-urlencoded" },
    body: payload.toString(),
  });

  if (!response.ok) {
    throw new Error(`HTTP ${response.status}: ${response.statusText}`);
  }

  const data = await response.json();

  for (const entity of data) {
    if (entity.success) {
      const info = entity.payload.user_information ?? {};
      console.log(`Titular: ${info.name ?? "N/A"}`);

      for (const acc of entity.payload.accounts ?? []) {
        console.log(`  Cuenta: ${acc.code}  Saldo: ${acc.balance ?? "N/A"} ${acc.currency ?? ""}`);
      }
      for (const port of entity.payload.portfolios ?? []) {
        console.log(`  Cartera: ${port.name}  Valor: ${port.total_value ?? "N/A"} ${port.currency ?? ""}`);
      }
    } else {
      console.error(`Error: ${entity.message ?? "desconocido"}`);
    }
  }
}

getEntities().catch(console.error);

File: examples/wealthreader.py
"""
Wealthreader API - Ejemplo en Python
Obtiene activos financieros de una entidad bancaria.
Requiere: pip install requests
"""

import requests

API_URL = "https://api.wealthreader.com/entities/"

payload = {
    "api_key": "TU_API_KEY",
    "code": "caixabank",
    "token": "TU_TOKEN",
    "product_types": "accounts,portfolios",
    "date_from": "2024-01-01",
}

response = requests.post(API_URL, data=payload)
response.raise_for_status()

data = response.json()
for entity in data:
    if entity.get("success"):
        info = entity["payload"].get("user_information", {})
        print(f"Titular: {info.get('name', 'N/A')}")

        for account in entity["payload"].get("accounts", []):
            print(f"  Cuenta: {account['code']}  Saldo: {account.get('balance', 'N/A')} {account.get('currency', '')}")

        for portfolio in entity["payload"].get("portfolios", []):
            print(f"  Cartera: {portfolio['name']}  Valor: {portfolio.get('total_value', 'N/A')} {portfolio.get('currency', '')}")
    else:
        print(f"Error: {entity.get('message', 'desconocido')}")
